### PR TITLE
feat(NumberFormat): add "compact" property – format a number compact with an abbreviation

### PR DESCRIPTION
**Pull request by tujoworker**
- It supports both plain numbers and `currency`.
- It supports easy decimals usage, see codesanbox above

Here's a Codesandbox which compares the usage usage and implementaion of decimals: https://codesandbox.io/s/eufemia-starter-compact-numbers-623fc0?file=/src/App.tsx

A11y support is included by providing the "long" version:
Screenshot 2022-03-01 at 10 13 18

### CODE REVIEW

File: packages/dnb-eufemia/src/components/number-format/NumberUtils.d.ts [hunk 1]
```diff
@@ -34,6 +34,9 @@ export interface formatOptionParams {
   /** Should the number be cleaned */
   clean?: boolean;
 
+  /** shortens any number including an abbreviation. It gives you zero decimal by default `decimals={0}`. You can combine `compact` with `currency`. Use either a boolean, or a string with "short" or "long" */
+  compact?: string | boolean;
```
Comment (thaytharma) [inline packages/dnb-eufemia/src/components/number-format/NumberUtils.d.ts:38]: https://github.com/dnbexperience/eufemia/pull/1324/files#r816628661

File: packages/dnb-eufemia/src/components/number-format/NumberFormat.js [hunk 2]
```diff
@@ -45,6 +45,9 @@ export default class NumberFormat extends React.PureComponent {
     currency_display: PropTypes.string,
     currency_position: PropTypes.oneOf(['auto', 'before', 'after']),
 
+    // shortens any number including an abbreviation
+    compact: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
```
[the remark this answers is not in the document]
Comment (tujoworker) [reply packages/dnb-eufemia/src/components/number-format/NumberFormat.js:49]: Yes, thanks!

Screenshot 2022-03-01 at 12 46 09

File: packages/dnb-design-system-portal/src/docs/uilib/components/number-format/Examples.js [hunk 3]
```diff
@@ -66,6 +66,37 @@ export const NumberCurrency = () => (
...
+    value={12345}
+  />
+  <NumberFormat
+    compact
+    currency
+    decimals={2}
+    value={-1723967.38}
+  />
+  <NumberFormat
+    compact
```
Comment (thaytharma) [inline packages/dnb-design-system-portal/src/docs/uilib/components/number-format/Examples.js:87]:
```suggestion
    compact="short"
```

Just a suggestion to see an example of "short" version as well.

**Review by thaytharma — APPROVED** [review]
Looks good!

File: packages/dnb-design-system-portal/src/docs/uilib/components/number-format/Examples.js (hunk 3 above)
Comment (tujoworker) [reply packages/dnb-design-system-portal/src/docs/uilib/components/number-format/Examples.js:87]: ahh, det short version is the default – will make it more clear in the docs 👍

**Pull request merged**